.build/images/dietpi-installer

```diff
@@ -322,30 +322,43 @@ _EOF_
 		G_DIETPI-NOTIFY 2 "Entered pre-image info: $PREIMAGE_INFO"
 
 		# Hardware selection
+		# if running in a container or vm, allow armv7 if on arm64
+		NON_NATIVE=0
+		hash systemd-detect-virt 2> /dev/null && systemd-detect-virt -q && NON_NATIVE=1
+		G_DIETPI-NOTIFY 2 "NON_NATIVE: $NON_NATIVE"
+		if hash systemd-detect-virt 2> /dev/null ; then
+			OUTPUT=$(systemd-detect-virt)
+		       	G_DIETPI-NOTIFY 2 "found the systemd-detect-virt command; output: $OUTPUT"
+		else
+		       	G_DIETPI-NOTIFY 2 "failed to find the systemd-detect-virt command"
+		fi
+
+		ARMV7_BOARDS=(
+			'' '●─ ARMv7 '
+			'10' ': Odroid C1'
+			'11' ': Odroid XU3/XU4/MC1/HC1/HC2'
+			'13' ': Odroid U3'
+			'70' ': Sparky SBC'
+			'52' ': ASUS Tinker Board'
+			'59' ': ZeroPi'
+			'60' ': NanoPi NEO'
+			'64' ': NanoPi NEO Air'
+			'63' ': NanoPi M1'
+			'66' ': NanoPi M1 Plus'
+			'48' ': NanoPi R1'
+			'61' ': NanoPi M2/T2/Fire2'
+			'25' ': Generic Allwinner H3'
+			)
 		# - NB: PLEASE ENSURE HW_MODEL INDEX ENTRIES MATCH dietpi-obtain_hw_model and dietpi-survey_report
 		# - NBB: DO NOT REORDER INDICES. These are now fixed and will never change (due to survey results etc)
 		G_WHIP_BUTTON_CANCEL_TEXT='Exit'
 		G_WHIP_DEFAULT_ITEM=0
 		case $G_HW_ARCH in
 			1) G_WHIP_MENU_ARRAY=('0' ': Raspberry Pi (all models)');;
-			2) G_WHIP_MENU_ARRAY=(
-				'0' ': Raspberry Pi 2 - 4'
-				'' '●─ ARMv7 '
-				'10' ': Odroid C1'
-				'11' ': Odroid XU3/XU4/MC1/HC1/HC2'
-				'13' ': Odroid U3'
-				'70' ': Sparky SBC'
-				'52' ': ASUS Tinker Board'
-				'59' ': ZeroPi'
-				'60' ': NanoPi NEO'
-				'64' ': NanoPi NEO Air'
-				'63' ': NanoPi M1'
-				'66' ': NanoPi M1 Plus'
-				'48' ': NanoPi R1'
-				'61' ': NanoPi M2/T2/Fire2'
-				'25' ': Generic Allwinner H3'
-				'' '●─ ARMv8 '
-			);;
+			2) G_WHIP_MENU_ARRAY=('0' ': Raspberry Pi 2 - 4')
+				G_WHIP_MENU_ARRAY+=(${ARMV7_BOARDS[@]})
```

Comment:
This would be needed to satisfy shellcheck:
```suggestion
				G_WHIP_MENU_ARRAY+=("${ARMV7_BOARDS[@]}")
```
Otherwise the spaces within the array entries would split them. You should be able to see the issue when you actually test the script with your change, having menu entires like:
```
0 :
Raspberry Pi
2 -
4
●─ ARMv7
```
etc. Bash btw does not take the whole array as single argument/entry when double quoting it as long as it is called with `${array[@]}`, but the quotation is interpreted a way that each element of the original array is quoted and hence treated as individual argument/entry.

However, I actually have a better idea to solve our issue.